### ISSUE TITLE
oemof licence

### ISSUE THREAD

**Issue by Ludee** [oemof/oemof#9]
While working on the open_eGo / oedb licence concept I had a look at the oemof licence and its application in the code:

As you know it is published under [GPL-3.0](https://opensource.org/licenses/gpl-3.0.html). -> [TLDR](https://tldrlegal.com/license/gnu-general-public-license-v3-%28gpl-3%29)

Because your [LICENCE](https://github.com/oemof/oemof/blob/dev/LICENSE) is just a copy of the "original" some of the required information are not included yet! Here is the full list:

**You Must:**
- **Include Original** (Describes whether copies of the original software or instructions to obtain copies must be distributed with the software.)
- **State Changes** (Stating significant changes made to software.)
- **Disclosure Source** (All code linked with GPL 3.0 source code must be disclosed under a GPL 3.0 compatible license.)
- **Include License** (Including the full text of license in modified software.)
- **Include Copyright** (Describes whether the original copyright must be retained.)
- **Include Install Instruction** (If the software is part of a consumer device, you must include the installation information necessary to modify and reinstall the software.)

To conclude, there is no general author defined and should be stated [here](https://github.com/oemof/oemof/blob/dev/LICENSE#L634-L635)
[Authors I found](https://github.com/oemof/oemof/search?utf8=%E2%9C%93&q=author): no author, author = u'oemof-Team', author='oemof developing group', author: uwe
Following the licence I will have to list all stated authors ;)

Feel free to check the boxes if you think it's done!

**Comment by Ludee** [comment]
General discussion in open_eGo https://github.com/openego/data_processing/issues/27

**Comment by ckaldemeyer** [comment]
> General discussion in open_eGo openego/data_processing#27

Shouldn't the discussion take place here as different projects are affected?

Thanks for bringing up this topic! Concerning your "TODO-List":

I guess someone should make a suggestion on how to fix this. It might also be a topic for the developer meeting (https://github.com/oemof/oemof/issues/218).

**Comment by Ludee** [comment]
I agree that the general discussion should be here because it affects the projects [open_eGo](https://github.com/openego) and [open_FRED](https://github.com/orgs/open-fred/dashboard).

There is an WebConf with focus on eGo on Monday (5.12.2016 13:00) if you like to join @all.
The goal is to define the next necessary steps for a proper license system.

**Comment by ckaldemeyer** [comment]
> I agree that the general discussion should be here because it affects the projects open_eGo and open_FRED.

> There is a WebConf with focus on eGo on Monday (5.12.2016 13:00) if you like to join @all.
> The goal is to define the next necessary steps for a proper license system.

I probably won't be able to make it. Could you post the findings here?

**Comment by ckaldemeyer** [comment]
The license file is still from the plain license and includes instructions, ...: https://github.com/oemof/oemof/blob/dev/LICENSE

We should adapt it and I'll add a TODO in https://github.com/oemof/oemof/issues/382

**Comment by p-snft** [comment]
Actually, the LICENSE file should never be modified ([GPL, line 6](https://github.com/oemof/oemof/blob/dev/LICENSE#L6)). Instead, the individual files should contain a copyright hint. What @Ludee noted, was a description how to do this, including a template. In fact, most of your files already contain such a text:

> '''
> This file is part of project oemof (github.com/oemof/oemof). It's copyrighted
> by the contributors recorded in the version control history of the file,
> available from its original location oemof/oemof/energy_system.py
> SPDX-License-Identifier: GPL-3.0-or-later
> '''

This is not the proposed text, but consider that it was formulated more than 10 years ago, when version control was not broadly used as distribution channel. Still, you might want to add some lines containing a more human-readable alternative to the SPDX code:

> oemof is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. See https://www.gnu.org/licenses/gpl-3.0.en.html for a full copy of the license.

All other points should automatically be fulfilled by publishing the files including the header quoted above.

**Comment by Ludee** [comment]
I got this mail from Robbie, who is an expert on OS-licenses. Looks like our choice for REUSE + GIT is not a suitable option.

> Because of the proposed JuMP rewrite, I was looking at the way oemof deploys GPLv3+ license notices.  oemof, I guess, read the FSFE REUSE guidelines and opted to use git for logging contributors.  And then copied boilerplate from the REUSE documentation into the top of python source files to action the method.

> But unfortunately the REUSE guidelines (version 2.0) are very poorly written on this particular topic.  I have been talking to Jonas Oberg <jonas@fsfe.org>, who runs the REUSE project, about clarifying or even removing the material on the use of version control systems (VCS) to handle copyright attribution. If you read the current guidelines carefully, there are still lots of caveats about the practice.

> The short answer is that VCSs are not recommended and that projects should manage copyright attribution by hand instead.
> I have quite a lot of traffic on this issue, so if you need more background, just ask.
> You are also welcome to forward this email as you see fit.

**Comment by p-snft** [comment]
Having read this: You are right, relying on the VCS can be problematic. An obvious case is when the service is discontinued. But also when somebody copies third-party code of which a copyright owner needs to be named. A typical (and easy) approach might be to have a CONTRIBUTORS file in the project root directory. The header then can point to this file instead of the VCS.

**Comment by p-snft** [comment]
After all these considerations, I find the approach we now follow e.g. in [https://github.com/Inovesa/Inovesa/blob/develop/src/IO/ProgramOptions.cpp](Inovesa) pretty manageable:

    // SPDX-License-Identifier: GPL-3.0-or-later
    /*
     * Copyright (c) Author A
     * Copyright (c) Author B
     * Copyright (c) Institution A
     */

Here, every author adds her name as a line to the file she edits. This allows some tracking of the copyright on the file level (not on a per-line basis, but this is cumbersome without version control). The license statement is not in the (long) legally binding form, but that is done at a central place to not extend the headers by too much.

**Comment by Ludee** [comment]
There is a new version (3.0) of the REUSE project:
https://reuse.software/spec/
This may of interest as well.

**Comment by p-snft** [comment]
I found that REUSE wants the license to be located at e.g. _./LICENSES/GPL-3.0-or-later.txt_ or _./LICENSES/MIT.txt_. As this is different from what people expect for single-license projects (_./LICENSE_, which is also GitHubs default), I would not do this.

However, I would still like the idea to name all contributors in the corresponding files.

**Comment by busiing** [comment]
Is this still relevant? If not - I will close this issue :)

**Comment by uvchik** [comment]
In oemof-solph we tried to follow the reuse standard. Due to the restructuring some history data was not kept, so we decided to use the author names.

The names are in random order but everybody is free to change it to alphabetic order.

I used the name of the github profile. Now it is easy to change the information using find/replace.

see: https://github.com/oemof/oemof-solph/blob/dev/src/oemof/solph/components.py

If you like we could use it as a template for other oemof repositories.
